stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -425,12 +482,143 @@ extension UnsafeMutableBufferPointer {
   ///   buffer, and an index to the point in the buffer one past the last
   ///   element written.
   @_inlineable
-  public func initialize<S: Sequence>(from source: S) -> (S.Iterator, Index)
-    where S.Element == Element {
+  @discardableResult
+  public
+  func initialize<S>(from source: S) -> (S.Iterator, Index)
+    where S:Sequence, S.Element == Element
+  {
     return source._copyContents(initializing: self)
   }
+
+  @_inlineable
+  public
+  func assign(repeating repeatedValue:Element)
+  {
+    guard let dst_start:UnsafeMutablePointer<Element> = self._position
+    else
+    {
+      return
+    }
+
+    dst_start.assign(repeating: repeatedValue, count: self._end! - dst_start)
+  }
+
+  @_inlineable
+  @discardableResult
+  public
+  func assign<S>(from source: S) -> (S.Iterator, Index)
+    where S:Sequence, S.Element == Element
+  {
+    return source._copyContents(assigningTo: self)
+  }
+
+  @_inlineable
+  public
+  func moveAssign(from source:UnsafeMutableBufferPointer<Element>)
+  {
+    _debugPrecondition(source.count >= self.count,
+                      "${Self}.moveAssign(from:) source does not have enough elements")
+    guard let dst_start:UnsafeMutablePointer<Element> = self._position,
+          let src_start:UnsafeMutablePointer<Element> = source._position
+    else
+    {
+      return
+    }
+
+    dst_start.moveAssign(from: src_start, count: source._end! - src_start)
+  }
+
+  @_inlineable
+  public
+  func moveInitialize(from source:UnsafeMutableBufferPointer<Element>)
+  {
+    _debugPrecondition(source.count >= self.count,
+                      "${Self}.moveInitialize(from:) source does not have enough elements")
+    guard let dst_start:UnsafeMutablePointer<Element> = self._position,
+          let src_start:UnsafeMutablePointer<Element> = source._position
+    else
+    {
+      return
+    }
+
+    dst_start.moveInitialize(from: src_start, count: source._end! - src_start)
+  }
+
+  @_inlineable
+  public
+  func deinitialize()
```

Comment:
Please add a scary warning comment to UnsafeMutableBufferPointer.deinitialize(). Something like:

Warning: All buffer elements must be initialized before calling this. Note that invoking `initialize(from:)` may not initialize all elements. Deinitializing part of the buffer must be done using the `UnsafeMutablePointer` API: `baseAddress.deinitialize(count: n)`